### DISCUSSION TITLE
Config problem with SATPI on Octagon SF4008 STB (Enigma2) + TVHeadend + Diseqc switch + 3x Quad LNB Setup

### DISCUSSION THREAD

**Discussion by gert1974** [Barracuda09/SATPI#202] [Q&A] [answered]
Hi,

I recently discovered the SatPI project and I think it is really interesting.
But I'm probably doing something wrong as I can't get any signal from 1 of my LNB's (Astra 1). The other 2 do work.

My setup =
1 Satellite Dish -> 3x Quad LNB (Astra 2 - Astra 3 - Astra 1 ) -> 4x Diseqc swicthes -> 4 receivers

Receiver 1 = Mutant HD51 (Enigma2), works perfectly
Receiver 2 = Raspberry Pi + DVBSky USB receiver + TVHeadend, works perfectly
Receiver 3 = Raspberry Pi + DVBSky USB receiver + TVHeadend, works perfectly
Receiver 4 = Octagon SF4008 (Enigma 2), works perfectly (apart from the remote control that went missing)

I am now trying to setup SATPI on the Octagon SF4008 and feed this to yet another Raspberry Pi with TVHeadend installed on it.

-

The Octagon SF4008 is running OpenEight 6.8.1 with SATPI installed on it.
It has a DVB-S2 tuner and a DVB-C/T tuner (not connected).

The DVB-S2 tuner is Frontend2 in SATPI. 
I selected DiSEqC switch in the configuration for Frontend2 but left every other value at its defaults.

My TVHeadend on the new Raspberry Pi detects the SATPI server.
I enabled Tuner1 and Postions 1, 2 and 3 and assigned them to the correct networks.
Position 1 = AA = Astra2
Position 2 = AB = Astra 3
Position 3 = BA = Astra 1
Position 4 = BB = < disabled>, not connected 

(This is similar to the setup of the TVHeadend setup with the DVBSky USB-receiver)

In TVH I left Tuner 2 (DVB-S) and all others (4x DVB-T and 2x DVB-C)  alone and disabled.

I can tune to and record anything I want from Astra 2.
I can tune to and record anything I want from Astra 3.
I can not tune to anything on Astra 1.

On the TVHeadend connected to the SATPI server I removed all discovered muxes and services and started from scratch.
All the muxes and services for Astra 2 & 3 get detected and are working (I can tune and record). I get no results for Astra 1 once again.

I am sure all the hardware is working. I switched the receivers around and it's always the Octagon with SATPI that does not work on Astra 1.

So, I most likely did something wrong in the configuration but I just don't see it.
Can you help me out?

**Comment by Barracuda09** [comment]
Did you setup in SatPI, in the WebUI in frontend tab, the correct DiSEqC Configuration?

You should setup there also DiSEqC switch

**Comment by gert1974** [comment]
> Did you setup in SatPI, in the WebUI in frontend tab, the correct DiSEqC Configuration?
> 
> You should setup there also DiSEqC switch

I selected DiSEqC switch in the configuration for Frontend2 but left every other value at its defaults.
Is that what you meant? Or is there anything else I need to configure?

There don't seem to be much more options I can change?

I see only 
Lof High (MHz)  
Lof Low (MHz)
Lof Switch (MHz)

for the LNB's.
All are at 
10600
9750
11700

Or am I not looking in the correct place?

**Comment by Barracuda09** [comment]
Could you please attach a picture of the `DiSEqC Configuration` and `LNB` part of the frontend config page?

which type of DiSEqC switch did you select

**Comment by Barracuda09** [comment]
So to be sure I understand the problem a few questions:

- One switch is also connected to the receiver with SatPI? with the (Astra 1 + 2 + 3)
- Only switch position 3 is not working? or none at all.

**Reply by gert1974** [reply]
> So to be sure I understand the problem a few questions:
> 
>     * One switch is also connected to the receiver with SatPI? with the (Astra 1 + 2 + 3)

Correct.
Every LNB has 4 outputs. I have 4 Diseqc switches (one for every receiver)


>     * Only switch position 3 is not working? or none at all.

Correct. Astra 2 (pos 1) and Astra 3 (pos 2) are working.
Pos 3 (Astra 1) does not work.
Pos 4 is empty.

**Comment by gert1974** [comment]
FrontEnd2 is my DVB-S tuner

![Screenshot 2023-12-12 153009](https://github.com/Barracuda09/SATPI/assets/5542215/e3c66c7a-d849-4a4f-8fb2-987f03d16cb8)

I changed the timeout and picked "Diseqc Switch" 

![Screenshot 2023-12-12 153158](https://github.com/Barracuda09/SATPI/assets/5542215/0efec8b3-606e-49cf-841e-f69779751035)

These are the LNB configurations (I left them alone)

![Screenshot 2023-12-12 153316](https://github.com/Barracuda09/SATPI/assets/5542215/b3af9a5e-ee20-49ed-80af-a7986cf38125)

![Screenshot 2023-12-12 153405](https://github.com/Barracuda09/SATPI/assets/5542215/835a1b92-6d20-4fdf-9cfe-90149cdc7c6d)

**Comment by Barracuda09** [comment]
Which version of SatPI is this? probably not a current one?

**Reply by gert1974** [reply]
> Which version is this? probably not a current one?

You mean the version of SatPI? 
That's the one I got from the repository of my OpenEight on my Octagon SF4008.
That distribution itself is up-to-date, I made sure of that.

This is the info I get from the package itself
![Screenshot 2023-12-12 154943](https://github.com/Barracuda09/SATPI/assets/5542215/611df2cf-2c3f-41e6-b331-586d0d58e9e3)


And on the web interface
![Screenshot 2023-12-12 155020](https://github.com/Barracuda09/SATPI/assets/5542215/c33340e8-bda5-46c8-9ab7-135acd9c3dee)



Am I missing some settings here and there???

**Reply by Barracuda09** [reply]
Yes, this is a old version of SatPI. You should have a version of something like 1.6.2.161~ga9aab2c

**Reply by gert1974** [reply]
I see. 

And those new versions have options that I do not have but need in order to get it configure correctly... ?

**Reply by Barracuda09** [reply]
Well this version is very old, and probably has bugs that are fixed in the meantime. And has some more DiSEqC switch options.

Do you have some experience in Linux and how to replace manually the SatPI  bin? and do you know which cpu the receiver has? ARM MIPS, MIPSEL?

**Reply by gert1974** [reply]
Right. That would probably explain a lot.

I have some experience with Linux. I know how to use PuTTy and SSH and can make config changes etc...
I know how to replace files and things like that.

Making my own build (like I see on the Wiki) would probably be a bridge too far, though.

![Screenshot 2023-12-12 163119](https://github.com/Barracuda09/SATPI/assets/5542215/2cbf5108-5969-41f1-aa18-b3c5efcf2979)


root@sf4008:~# cat /proc/cpuinfo
processor       : 0
model name      : ARMv7 Processor rev 3 (v7l)

**Reply by Barracuda09** [reply]
I can cross that bridge for you.

Here is a new one [satpi_ARM_1.6.2.161~ga9aab2c_Enigma.zip](https://github.com/Barracuda09/SATPI/files/13651166/satpi_ARM_1.6.2.161.ga9aab2c_Enigma.zip)

You need to update the web folder as well, just copy that from here https://github.com/Barracuda09/SATPI/tree/master/web

**Reply by gert1974** [reply]
I updated the SATPi binary and the web folder. That was easy, thanks!

I am now running the coreect version. 
I can confirm from the webpage: 
Version | 1.6.2.161~ga9aab2c Enigma



However, still no luck with Pos 3 / Astra 1.

I have now a few more options, indeed. 
I have selected Diseqc (like before). 

I have switched around between Committed and Uncommitted as Switch type and between 3 and 4 as the number of switch inputs.
None of that seems to make a difference.
Pos 1 / Astra 2 and Pos 2 / Astra 3 both work, whatever I do.
Pos 3 / Astra 1 never works, wahtever I do.

(I left all other settings as they were)


I am probably doing something wrong, but right now I have no clue what...

**Reply by Barracuda09** [reply]
Yes, the switch is most likely an commited switch.

Could you try something with VLC this is Tagesschau 24 HD on Astra1 (Change IP adress to your server)
```
rtsp://192.168.1.15/?src=3&fe=3&freq=11052.75&sr=22000&msys=dvbs2&mtype=8psk&pol=h&fec=23&pids=0,1,16,17,18,5400,5401,5402,5404
```

**Reply by gert1974** [reply]
No luck with that either.

I tried the playlist and the only thing from the "Auto"playlist that works for me is the Canal Digitaal Infokanaal.
I guess all except the Germans from Astra1 are encrypted.


I tried ZDF, CNBC Europe and ARD-alpha. They should all play I think but none do.

**Reply by Barracuda09** [reply]
Could you please give a log, when making the previous test with VLC.
In the log the SETUP message should be visible.

**Reply by Barracuda09** [reply]
You can also enable in WebUI `Configure` tab `Log messages to syslog` so you can zip that and attach it here

**Reply by gert1974** [reply]
I'll get back to that, but probably not before Sunday.
I have some family business to take care of.

I also think I screwed up my SatPI installation while trying a few things out.
Right now not even Astra2 or Astra3 are working.
I need to get that back to a working state first (or maybe start from scratch)


I'll get back to this!

**Reply by Barracuda09** [reply]
Ok, no problem.

**Comment by gert1974** [comment]
Hi Marc

I have attached a log file (a copy of all the entries in LOG)

First entries (12:33 - 12:36) are a recording from 'Canvas' via TVHeadend on a resperry pi. That worked.
Last entries (12:51 - 12:55) are a VLC connection from my laptop to the link you provided (Tagesschau 24 HD). That did not work. I just got a black screen in VLC.

[20231217_satpi.log](https://github.com/Barracuda09/SATPI/files/13695929/20231217_satpi.log)

**Comment by Barracuda09** [comment]
What type/brand of LNB do you use?

**Reply by gert1974** [reply]
Frankly, I have no idea about the brand/make or type. 
There are no insriptions or even stickers on them.

I bought them secondhand, together with the Satellite Dish and 3 of the Diseqc switches from an ex-colleague.
It was a working set on the hosue he bought at the time but he wasn't interested in thema nd I bought them from him.

**Comment by Barracuda09** [comment]
Did you try to increase `Repeat DiSEqC` to 1
Strange that src=3 does not work, is does not lock around 12:51:13.3491
```
Sun Dec 17 12:51:11.0071 2023 6 RTSP Connection from 172.27.64.75 Port 50411 with fd: 8
 Sun Dec 17 12:51:11.0294 2023 7 RTSP Stream data from client None with IP 172.27.64.75 on Port 50411: SETUP rtsp://172.27.64.87:554/?src=3&fe=3&freq=11052.75&sr=22000&msys=dvbs2&mtype=8psk&pol=h&fec=23&pids=0,1,16,17,18,5400,5401,5402,5404 RTSP/1.0
 Sun Dec 17 12:51:11.0294 2023 7 CSeq: 0
 Sun Dec 17 12:51:11.0294 2023 7 Transport: RTP/AVP;unicast;client_port=9358-9359
 Sun Dec 17 12:51:11.0294 2023 7 <CRLF>
 Sun Dec 17 12:51:11.0295 2023 6 Found FrondtendID: 3 (fe=3)  StreamID: -1  SessionID: 0263495685
 Sun Dec 17 12:51:11.0296 2023 7 Frontend: 3, Found Streaming type: RTP/AVP
 Sun Dec 17 12:51:11.0296 2023 6 Frontend: 3, StreamClient with SessionID 0263495685 for dvbs2
 Sun Dec 17 12:51:11.0297 2023 6 Frontend: 3, Parsing transport parameters...
 Sun Dec 17 12:51:11.0297 2023 6 Frontend: 3, New frequency requested (0.0000 -> 11052.7500), clearing old channel data...
 Sun Dec 17 12:51:11.0299 2023 7 Frontend: 3, Parsing PID parameter: pids=0,1,16,17,18,5400,5401,5402,5404
 Sun Dec 17 12:51:11.0299 2023 7 Frontend: 3, Parsing transport parameters (Finished)
 Sun Dec 17 12:51:11.0300 2023 6 Frontend: 3, Updating frontend...
 Sun Dec 17 12:51:11.0300 2023 6 Frontend: 3, Closing all active PID filters...
 Sun Dec 17 12:51:11.0804 2023 6 Frontend: 3, Opened /dev/dvb/adapter0/frontend2 for Read/Write with fd: 9 (0 ms)
 Sun Dec 17 12:51:11.0804 2023 6 Frontend: 3, Start tuning process for DVB-S(2)...
 Sun Dec 17 12:51:11.0805 2023 6 Frontend: 3, Opened /dev/dvb/adapter0/frontend2 for Writing DiSEqC command with fd: 9
 Sun Dec 17 12:51:11.0807 2023 7 Frontend: 3, Using LNB with: Lof-High=10600 MHz  Lof-Low=9750 MHz  Lof-Switch=11700 MHz -> IF=1302 MHz
 Sun Dec 17 12:51:11.0807 2023 6 Frontend: 3, Sending DiSEqC: [0xE0] [0x10] [0x38] [0xFA] - DiSEqC Src: 2
 Sun Dec 17 12:51:11.3083 2023 7 Frontend: 3, Set Properties: Frequency 1302750
 Sun Dec 17 12:51:11.3084 2023 6 Frontend: 3, Tuned, waiting on lock...
 Sun Dec 17 12:51:13.3491 2023 6 Frontend: 3, Not locked yet   (FE status 0x20)...    <*****************
 Sun Dec 17 12:51:13.3492 2023 6 Frontend: 3, Not locked yet   (Timeout 2268 ms)...
 Sun Dec 17 12:51:13.3492 2023 6 Frontend: 3, Updating PID filters...
 Sun Dec 17 12:51:13.3493 2023 6 Frontend: 3, Opened /dev/dvb/adapter0/demux2 using fd: 10
 Sun Dec 17 12:51:13.3495 2023 6 Frontend: 3, Set DMX buffer size to 3145728 Bytes
 Sun Dec 17 12:51:13.3497 2023 6 Frontend: 3, Set DMX_SET_SOURCE with (Src: 2 - Offset: 0)
 Sun Dec 17 12:51:13.3711 2023 7 Frontend: 3, Set filter PID: 0000
 Sun Dec 17 12:51:13.3913 2023 7 Frontend: 3, Set filter PID: 0001
 Sun Dec 17 12:51:13.4114 2023 7 Frontend: 3, Set filter PID: 0016
 Sun Dec 17 12:51:13.4315 2023 7 Frontend: 3, Set filter PID: 0017
 Sun Dec 17 12:51:13.4517 2023 7 Frontend: 3, Set filter PID: 0018
 Sun Dec 17 12:51:13.4719 2023 7 Frontend: 3, Set filter PID: 5400
 Sun Dec 17 12:51:13.4921 2023 7 Frontend: 3, Set filter PID: 5401
 Sun Dec 17 12:51:13.5123 2023 7 Frontend: 3, Set filter PID: 5402
 Sun Dec 17 12:51:13.5324 2023 7 Frontend: 3, Set filter PID: 5404
 Sun Dec 17 12:51:13.5325 2023 6 Frontend: 3, Updating frontend (Finished in 2502 ms)
 Sun Dec 17 12:51:13.5326 2023 6 Frontend: 3, RTP/UDP set network buffer size: 4096 KBytes
 Sun Dec 17 12:51:13.5326 2023 6 Frontend: 3, Start RTP/UDP stream to 172.27.64.75:9358
 Sun Dec 17 12:51:13.5326 2023 6 Frontend: 3, Start RTCP/UDP stream to 172.27.64.75:9359
 Sun Dec 17 12:51:13.5328 2023 7 Frontend: 3, Start Reader and Monitor Thread
 Sun Dec 17 12:51:13.5328 2023 7 Send reply in 2503 ms
 Sun Dec 17 12:51:13.5328 2023 7 RTSP/1.0 200 OK
 Sun Dec 17 12:51:13.5328 2023 7 Server: satpi/1.6.2.161~ga9aab2c Enigma
 Sun Dec 17 12:51:13.5328 2023 7 CSeq: 0
 Sun Dec 17 12:51:13.5328 2023 7 Session: 0263495685;timeout=60
 Sun Dec 17 12:51:13.5328 2023 7 Transport: RTP/AVP;unicast;client_ip=172.27.64.75;client_port=9358-9359
 Sun Dec 17 12:51:13.5328 2023 7 com.ses.streamID: 102
 Sun Dec 17 12:51:13.5328 2023 7 <CRLF>

```

**Comment by Barracuda09** [comment]
Could you try this test version:
[satpi_ARM_1.6.2.161~TEST_DISEQC_Enigma.zip](https://github.com/Barracuda09/SATPI/files/13697928/satpi_ARM_1.6.2.161.TEST_DISEQC_Enigma.zip)

**Reply by gert1974** [reply]
The test version itself did not change anything.
(This version also identified itself as 1.6.2.161~ga9aab2c Enigma. Is that expected or did I do something wrong?)

-

However, this made it work!

>> Did you try to increase Repeat DiSEqC to 1

I changed Repeat Diseqc from 0 to 1, rebooted the Octagon receiver and after that 
* I could play "Tageschau 24HD" with VLC
* I could record something from "RTL Television" on TVHeadend on the RaspberryPi.
* Astra 2 and 3 also still work, I tried BBC Two and RTL7, both record OK.


Is it the new version that made it work or "just" setting Diseqc repeat to 1 instead of 0 ?

-

I checked the configs of my TVHeadend where I have an USB DVS2-adapter attached. 
There I have "Diseqc repeats = 0"
So I am unsure why a different setting is needed for SATPi.


But if it keeps working, that would be great!

**Reply by Barracuda09** [reply]
I do not exactly know what did the trick at this point. it is a bit strange because source 2 worked so the Diseqc command did arrive. Source 1 is probably the default that does not need switching around.

> The test version itself did not change anything.
> (This version also identified itself as 1.6.2.161~ga9aab2c Enigma. Is that expected or did I do something wrong?)

Yes this is not handy, but correct. 

it could be that the timing is a bit off, you could try:
- Do you know the brand of the switches?
- This version **EDIT add new version**  [satpi_ARM_1.6.2.170~g016343c Enigma.zip](https://github.com/Barracuda09/SATPI/files/13725713/satpi_ARM_1.6.2.170.g016343c.Enigma.zip).
- Set `Repeat DiSEqC` back to 0
- Increase in steps, the `Delay before write` and `Delay after write` with 5 ms and see when this works.

**Reply by gert1974** [reply, marked as answer]
> Do you know the brand of the switches?

I have 3 that are the same (they came with the set I bought - the previous owner only used 3 of the 4 outputs on the LNBs)
The inscription on it:

DIGIALITY 4 x 1 DISEqC Switch - DISEqC 1.0 
Frequency Range 950-2200MHz
Insertion Loss:<3dB
Power:12-20V/30mA


I bought the 4th one later: 
EMP-CENTAURI 
DISEqC 2.0 Switch S4/1PCN-W1
(P.164-IW)
Frequency Range SAT 950-2300MHz


It doesn't matter which one I connect the receiver to, with just Diseqc=0 none of them are working.


> This version 

OK, I installed that version.

> Set Repeat DiSEqC back to 0

With that alone it is not working

> Increase in steps, the Delay before write and Delay after write with 5 ms and see when this works.

When I set delay before to 35 and delay after to 40 it is working. 
30 and 35 does not work.

**Reply by Barracuda09** [reply]
Ok, so if I understand correctly, the timing is a bit critical for your use case?

**Comment by gert1974** [comment]
> Ok, so if I understand correctly, the timing is a bit critical for your use case?

Yes, I think it seems like that.
I tried a few times to lower the delay agan, but it consistently stops working then, and starts working again when I put it back up to 35/40

**Reply by Barracuda09** [reply]
Maybe I have to increase the defaults

**Comment by gert1974** [comment]
> Maybe I have to increase the defaults

That is up to you I think?
If I am the only one of all your uses who has this problem, then maybe not? 

I can't tell if it would cause other problems for everyone else or if it wouldn't make a difference at all for them.

---

Just a question.

I can make my setup work in 2 different ways:
* increase the delay to 35/40 and keep DISEQc repeat to 0 (= leave as standard)
* leave the delay to standard and put DISEQc repeats to 1.

Which of the 2 is the best way to do it for now?

**Reply by Barracuda09** [reply]
Just increase the delays. I will increase the timing this will not affect that much, just make it more reliable. 

Thanks you for testing!

**Comment by gert1974** [comment]
> Just increase the delays. I will increase the timing this will not affect that much, just make it more reliable.

Thank you! 

Many thanks for helping me out. As far as I can tell, eveything is working well (stable) with these settings!
The only problems I had were cuased by too much rain and wind :)


Merry Christmas & Happy New Year...

**Reply by Barracuda09** [reply]
No Problem, Thank you for testing!

I have increased the defaults to 35 and 40 ms

And you too, have a Merry Christmas and a Good new year!
